src/lib/import-export/import/importers/importer.ts

```diff
@@ -61,13 +96,33 @@ export class DefaultImporter extends EventEmitter implements Importer {
 
 		this.emit( ImportEvents.IMPORT_DATABASE_START );
 		const sortedSqlFiles = [ ...this.backup.sqlFiles ].sort( ( a, b ) => a.localeCompare( b ) );
-		for ( const sqlFile of sortedSqlFiles ) {
+
+		let backupDbPath = '';
+		const hasCreateWithoutDrop = await this.hasCreateWithoutDrop( sortedSqlFiles );
+		if ( hasCreateWithoutDrop ) {
+			backupDbPath = await this.backupExistingAndCreateNewDatabase( rootPath, siteId );
+		}
+
+		await this.importSqlFiles( rootPath, server, sortedSqlFiles );
+
+		if ( backupDbPath ) {
+			await shell.trashItem( backupDbPath );
```

Comment:
What is the point of having `backupDbPath`?

Reply:
I don't feel confident in removing the database forever, what if the import fails and the user wants to go back to the previous original database?

Reply:
@sejas in such a case, why can't we trash it at the beginning? Currently, we rename it, and we don't do anything with the renamed path besides trashing it in the end.

What happens when you import MySQL dump to MySQL server, and it fails? Does it preserve the previous database state?

> The fact that we are backing up the database makes me wonder if we should do it for the entire site. I.e. before the import, we create a copy of the files/directories to be replaced and zip them into a file. This could be useful to restore the site to a previous point, as well as revert an import that results in breaking the site.

There are plenty of possible improvements that could be made, but I would focus on the current goal which is to reset the database before importing.

Reply:
> There are plenty of possible improvements that could be made, but I would focus on the current goal which is to reset the database before importing.

Sure, I didn't want to convey that we should do this now. I'll open a GitHub issue as a follow-up.

Reply:
Let's remove the database directly when the user imports a Jetpack backup. 👍